benchmark/README.md

```diff
@@ -0,0 +1,101 @@
+# benchmark
+
+This is the code used to benchmark `alfred-margaret`.
+The results of this benchmark are shown in [Channable blog article](https://www.channable.com/tech/how-we-made-haskell-search-strings-as-fast-as-rust).
+
+## Setup
+
+Running `nix-shell` in this directory drops you into a shell where the build tools `cargo`, `bazel`, `stack` and the required python packages are available.
+The first time you do this will likely take a while unless you've already downloaded those packages.
+
+## Running the Benchmarks
+
+Files in the `data` directory must follow this format:
+
+```
+Lorem
+sunt
+officia
+
+Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
+```
+
+The first part of this file defines the search terms (the "needles") `Lorem`, `sunt` and `officia`.
+The second part, after a blank line, defines the corpus to search in  (the "haystack").
+Currently, these files **must be encoded as UTF-16**.
+
+Each run of `benchmark.py` generates a pair for `.results` and `.stats` files.
+
+### Python
+
+Run the benchmark in this directory
+
+```
+./benchmark.py ./naive.py --prefix python
+```
+
+### Rust
+
+Compile the Rust implementation by running Cargo in the `rust` directory:
+
+```
+cargo build --release
+```
+
+Run the benchmark using the compiled binary:
+
+```
+./benchmark.py rust/target/release/acbench-rust --prefix rust
+```
+
+### Java
+
+Compile the Java implementation by running Bazel in the `java` directory:
+
+```
+bazel build :acbench_deploy.jar
+```
+
+Run the benchmark using the script generated by Bazel:
+
+```
+./benchmark.py java/bazel-bin/acbench --prefix java
+```
+
+### Haskell
+
+Compile the Haskell implementation by running Stack in the `haskell` directory:
+
+```
+stack build
+```
+
+Run the benchmark using the compiled binary:
+
+```
+./benchmark.py haskell/.stack-work/dist/*/Cabal-3.0.1.0/build/ac-bench/ac-bench --prefix alfred-margaret
```

Comment:
This should use `--prefix haskell` to match with the report command below:

Reply:
Doesn't actually seem fixed (what I mean is that the example call of `report.py` down in "Inspecting the Results" uses `haskell.stats` in its command line.